### DISCUSSION TITLE
Undo

### DISCUSSION THREAD

**Discussion by halfk1ng** [Budibase/budibase#1054] [Ideas]
An undo button. (with keyboard short cut ;)

-- preferably one that can undo more than once. I know it's a lot to cache, so rolling out as single undo at first is still a good MVP feature!

**Comment by joebudi** [comment]
This is essential as time goes on IMO. There's currently a lot on the roadmap, but I personally, feel this is a critical one for improving user experience.

**Comment by lukuser** [comment]
Hello, Is this feature available yet? I accidentally deleted a top level container that I was building for the past few days? Any way to recover this?

**Reply by mindplay-dk** [reply]
> I accidentally deleted a top level container that I was building for the past few days?

Ouch. Yeah, this is why we design applications with an undo stack from the get-go. It's the main reason I won't consider this product. 😐

**Reply by melohagan** [reply]
It's in the roadmap for the first quarter of next year! https://github.com/Budibase/budibase/projects/26

**Reply by mindplay-dk** [reply]
Subscribed to the issue - thanks :-)
